### ISSUE TITLE
Parser error with nested Ifs

### ISSUE THREAD

**Issue by Felalolf** [viperproject/gobra#93]
Peter encountered a parser bug in the attached file.
[example.txt](https://github.com/viperproject/gobra/files/5306509/example.txt)

**Comment by jcp19** [comment]
The bug still happens in this minimal example
```go
package main

func Eg(x int, y int) {
	if x == y {
		// Fails. Works if I add this statement: z := 0
		// also works if I put ';'
	} else {

	}
}
```

**Comment by jcp19** [comment]
Smaller still
```go
package main

func Eg(x int, y int) {
	if x == y {
		// Fails. Works if I add this statement: z := 0
		// also works if I put ';'
	}
}
```

**Comment by jcp19** [comment]
The problem seems to be caused in the expression parser: if the if condition is put inside parentheses, the code parses correctly.

```go
// parses correctly
if (x == y) {
}
```

**Comment by ArquintL** [comment]
@jcp19 are you investigating this issue or should I have a look?

**Comment by jcp19** [comment]
I had to switch contexts from this for a while. If you want, you can take a look at this. Otherwise I will get back to it later :)

**Comment by ArquintL** [comment]
fun fact `y {}` is currently parsed as a composite literal and hence parsing of `if x == y {}` fails

**Comment by Felalolf** [comment]
uff. That seems really difficult to fix.

**Comment by ArquintL** [comment]
I've started on branch `issue-93` but without much success so far. Commit 96ded8c444ea3e433f1763a1e08705215422c352 contains some of my (failed) attempts to integrate contextual information into the parser. Might be useful when revisiting this issue at a later point

**Comment by jcp19** [comment]
Fixed after #388
